onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -1418,6 +1418,31 @@ ORT_API_STATUS_IMPL(OrtApis::ReleaseAvailableProviders, _In_ char** ptr,
   return NULL;
 }
 
+ORT_API_STATUS_IMPL(OrtApis::At, _Inout_ OrtValue* value, size_t* location_values, size_t location_values_count,
+                   _Outptr_ void** out) {
+  TENSOR_READWRITE_API_BEGIN
+  //TODO: test if it's a string tensor
+  if (location_values_count != tensor->Shape().NumDimensions())
+    return OrtApis::CreateStatus(ORT_INVALID_ARGUMENT, "location dimensions do not match shape size");
+  std::vector<size_t> location(location_values_count);
+  for (size_t i = 0; i < location_values_count; i++) {
+    if (location_values[i] >= (size_t)tensor->Shape()[i])
+      return OrtApis::CreateStatus(ORT_INVALID_ARGUMENT, "invalid location range");
+    location[i] = location_values[i];
+  }
+  // assume data has row-major order
```

Comment:
Assume sounds like there is some ambiguity, the tensor data should always be stored in row-major order. Just remove the 'assume' part?

Reply:
I wrote that in case someone ever wondered where the algorithm came from, they would see that it matches the formula on [this](https://en.wikipedia.org/wiki/Row-_and_column-major_order#Address_calculation_in_general) page. I updated the comment so it's less ambiguous.